### ISSUE TITLE
Error when library database directory doesn't exist

### ISSUE THREAD

**Issue by simonvanderveldt** [beetbox/beets#3808]
### Problem
When the directory the library database is contained in doesn't exist yet beets throws an error:
```
error: database file /home/simon/.local/share/beets/library.db cannot not be opened: unable to open database file
```

Expected behavior:
Beets creates the required (parent) directories

Version info: Current master (ed18551c1c6d24158bfd63fd5269c245513f5855)

**Comment by wisp3rwind** [comment]
I think this is intended behaviour, since it is quite common to have a music collection including the database on an external disk. When beets is accidentally run when the external drive is disconnected, no such parent directories should be created. I don't think there's an easy solution to detect whether this is the case (except for adding a configuration option to indicate that the collection is removable, but I'd say that's not really an improvement over creating the folder manually).

Semi-related: https://github.com/beetbox/beets/issues/1934

**Comment by simonvanderveldt** [comment]
@wisp3rwind This is not about the music collection itself but about the database ("library", the key used in the config, isn't the clearest of names). But I guess people could have their database also on an external drive together with the music?

**Comment by wisp3rwind** [comment]
> @wisp3rwind Thanks for the reply. To be clear, unlike the linked issue this is not about the music collection itself but about the database ("library", the key used in the config, isn't the clearest of names). But I guess people could have their database also on an external drive together with the music collection?

Thanks for the clarification, and sorry for the belated reply! I did understand you correctly initially already. At least I have the db located next to my music collection. To be honest, I have no idea what is generally more common among beets users. So, I'd prefer not to create files/directories in that situation automatically.

> If that's really a thing that happens wouldn't it make more sense to show a message saying no database was found and ask if it should be created? When answering yes the missing directories can be created as well.

Adding a prompt in that situation sounds very reasonable, we could add that. If you (or anyone else) would open a PR, that'd be perfect!

**Comment by alicezou** [comment]
Hi, I am working on this. Here is my pull request.
#4327

**Comment by alicezou** [comment]
Fixed in #4327 
The issue probably can be closed. ;)

**Comment by sampsyo** [comment]
Indeed; thanks!!
